Sources/fx-upscale/FXUpscale.swift

```diff
@@ -51,21 +52,33 @@ import Upscaling
         }
 
         let outputSize = CGSize(width: width, height: height)
+        let outputCodec: AVVideoCodecType? = {
+            switch codec.lowercased() {
+            case "prores": return .proRes422
+            case "h264": return .h264
+            default: return .hevc
+            }
+        }()
 
-        let convertToProRes = (outputSize.width * outputSize.height) > (3840 * 2160) &&
-            !(formatDescription?.videoCodecType?.isProRes ?? false)
+        // Through anecdotal tests anything beyond 14.5K fails to encode for anything other than ProRes
+        let convertToProRes = (outputSize.width * outputSize.height) > (14500 * 8156)
```

Comment:
This is probably limited by the h264/hevc profile used and could be actually determined somehow, but hardcoded works fine for now